csharp-basic-topics/AwaitInForEachLoopCsharp/Tests/TaskWhenAllTest.cs

```diff
@@ -0,0 +1,25 @@
+﻿
+
+using AwaitInForEachLoop;
+
+namespace Tests
+{
+    [TestFixture]
+    public class TaskWhenAllInLoopTests
+    {
+        [Test]
+        public async Task GivenResultAsync_ThenProcessNumbersAsync()
+        {
+            // Arrange
+            var expectedNumbers = new List<int> { 10, 20, 30, 40, 50, 60, 70, 80, 90 };
+
+            // Act
+            var result = await TaskWhenAllInLoop.ResultAsync();
+
+            // Assert
+            Assert.IsNotNull(result);
+            CollectionAssert.AreEqual(expectedNumbers, result);
```

Comment:
As per the latest code changes, the result will be different than the input. We can just check the count of result collection in this case. This will prove that the method execution is completed once all the tasks was over.